src/foam/dao/jdbc/JDBCPooledDataSource.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * @license
+ * Copyright 2019 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.dao.jdbc;
+
+import foam.core.X;
+import org.apache.commons.dbcp2.*;
+import org.apache.commons.pool2.ObjectPool;
+import org.apache.commons.pool2.impl.GenericObjectPool;
+
+import foam.nanos.logger.Logger;
+
+import javax.sql.DataSource;
+import java.sql.DriverManager;
+
+public class JDBCPooledDataSource {
+
+  // Holds a reference to the connection pool ( .getConnection() )
+  protected DataSource dataSource_;
+  protected String poolName_;
+
+  public JDBCPooledDataSource(X x, String poolName) {
+    poolName_ = poolName;
+
+    try {
+      //Add code to get JDBCConnectionSpec properties
+      Object spec = x.get("JDBCConnectionSpec");
+
+      if( spec == null ) {
+        throw new java.sql.SQLException("JDBC connection information not found");
+      }
+
+      if ( !(spec instanceof JDBCConnectionSpec) ) {
```

Comment:
Just cast to JDBCConnectionSpec on line 30 and the system will throw the ClassCastException for you and this block can be removed.

Reply:
ok